### DISCUSSION TITLE
Remove TrackElement.RideType

### DISCUSSION THREAD

**Discussion by spacek531** [OpenRCT2/OpenRCT2#17580] [Feature requests] [answered]
Does anyone have any parks that use this? It is redundant, users cannot manipulate it, and it is consuming all available space in the TrackElement struct. I would like to see more flags, alt track types, and sidecar data over RideType.

**Comment by karst** [comment]
To add to this from the discussion that has been had over several discord channels and a github page I think it's smart to add to this that the feature would most likely become redundant when OpenGraphics is completed as full sprite sets of every tracktype would be provided. The basic use for this feature is that you can use elements from other tracks that are drawn. Another suggestion was to reduce the amount of bits that ride type uses up, but reducing it back to 1 byte would still make for the usage of 8 bits.

**Reply by spacek531** [reply]
ride_type_t is a uint16, it would be a lot of work to limit it to 255 just for this. I think it should go entirely.

**Comment by karst** [comment]
How about this solution. Rather than choosing between all available ride types you chose between all built rides ("Continue building as other ride") and you can continue building a ride as another already built ride (and auto merge) if you want. It's definitely not as elegant but it shouldn't take up 16 bits (as it uses already available features from the game) and would give about the same result (for a feature that is already achievable in other ways) ^^;

**Reply by spacek531** [reply]
Instead of locking into something like this, we should investigate a sidecar data struct.

**Reply by Broxzier** [reply]
These "other ways" are only supported because it happens to work, and come with downsides. Merged tracks cannot be opened normally, and cannot use block brakes. Traversing them backwards makes the vehicle crash. The goal of this field is to properly support track merging.

**Comment by duncanspumpkin** [comment]
Of course nothing uses it yet because we haven't added the ui for it. This is to replace ride merging so i see that as quite an important thing.

**Reply by spacek531** [reply]
Since nobody uses it, it's easy to remove, then we can strategize how more data can be added to track elements.

If merging were patched, then this feature would be necessary, but until it is, people are going to continue merging. RideType is redundant and taking up valuable space that is necessary to continue developing rides.

**Reply by duncanspumpkin** [reply]
This isn't getting in the way of adding more data to the structure and it's the simplest solution to doing track merging elegantly. All we do to add more data is create a new component and add a ref to the element. Move the less used items from the element to the component, done. Just because no one uses it at the moment doesn't mean it isn't of use. Removing it would be a massive task for little to no gain.

**Reply by X123M3-256** [reply]
I just don't really see why we need a solution to doing track merging elegantly when the old way will have to continue to be supported for backward compatibility, and track merging is a hack to start with. The underlying problem is that many track types are missing elements. I think we should really be planning for the OpenGraphics project to fix that by adding the meaning sprites, rather than making track merging a properly supported feature. The old method of merging still works, and will presumably have to continue working as I don't see an easy way to automatically convert parks that used it to the new system.

There are some ride types which actually do use multiple track styles that should be switchable without hacks, but I think it would be better to add a flag or two to specify alternate track styles in a similar way to the way chain lifts are handled. If we have it so that each ride type is associated with a single track style, I don't know how that would work when ride types objects are implemented. I really wouldn't want a ride type to have to be split across multiple ride type objects just because it features multiple track types.

> All we do to add more data is create a new component and add a ref to the element.

Where's the reference going to go? There's no space left in the tile element. Are you planning to make tile elements larger than 16 bytes?

**Reply by duncanspumpkin** [reply]
We would move some of the fields into the component to make room for the ref. Its not really an issue. I don't want to have lots of flags all over the place they are inflexible this field is a better solution it just needs some further fine tuning.

**Comment by spacek531** [comment, marked as answer]
In the discord Duncan suggested that an ECS model (i.e. the sidecar data) would solve the space issue. This is a more difficult route to pursue but better in the long run and would solve many issues with track element data limits.
